core/test/pazuzu_core/rest_test.clj

```diff
@@ -0,0 +1,22 @@
+(ns pazuzu-core.rest-test
+  (:require [clj-http.client :as client]
+            [clojure.test :refer :all]
+            [pazuzu-core.rest :as rest])
+  (:use clj-http.fake))
+
+(def correct-sorted-features
+  [{"name" "cool-feature-0", "docker_data" "RUN echo 1", "test_instruction" "RUN echo  \"test echo\""}
+   {"name" "cool-feature-2", "docker_data" "RUN echo 2", "test_instruction" "RUN echo  \"test echo\""}
+   {"name" "cool-feature-1", "docker_data" "RUN echo 1", "test_instruction" "RUN echo  \"test echo\""}])
+
+(def sorted-features-response-body
+  "[{\"name\":\"cool-feature-0\",\"docker_data\":\"RUN echo 1\",\"test_instruction\":\"RUN echo  \\\"test echo\\\"\"},{\"name\":\"cool-feature-1\",\"docker_data\":\"RUN echo 1\",\"test_instruction\":\"RUN echo  \\\"test echo\\\"\"},{\"name\":\"cool-feature-2\",\"docker_data\":\"RUN echo 2\",\"test_instruction\":\"RUN echo  \\\"test echo\\\"\"}]")
+
+(with-fake-routes {
```

Comment:
midje provides a generic solution for faking any function and checking if it was actually called and with which parameters.
```clj
(facts "about get-sorted-features"
  (rest/get-sorted-features "http://localhost:8080" ["cool-feature-2" "cool-feature-1"]) => correct-sorted-features
  (provided
    (clj-http.client/get "http://localhost:8080/api/features?sorted=1&name=cool-feature-2,cool-feature-1" anything)
    => {:status 200 :body sorted-features-response-body}))
```